### DISCUSSION TITLE
Define API server route in module

### DISCUSSION THREAD

**Discussion by green-cats** [nuxt/nuxt#17405] [Questions] [answered]
Hi!
Can I define API server route in my own module? 
Thanks!

**Comment by Diizzayy** [comment, marked as answer]
@green-cats Ofcourse, you can leverage `addServerHandler` function as seen below:

`module.ts`
```ts
import { defineNuxtModule, createResolver, addServerHandler, } from '@nuxt/kit'

export default defineNuxtModule({
  setup (opts, nuxt) {
    const { resolve } = createResolver(import.meta.url)

      addServerHandler({
        method: 'post',
        route: '/api/demo',
        handler: resolve('runtime/handlers/demo.ts')
      })
  }
})
```

`runtime/handlers/demo.ts`
```ts
export default defineEventHandler((event) => {
  return 'demo'
})

```

**Reply by green-cats** [reply]
@Diizzayy Thank you!

**Reply by green-cats** [reply]
@Diizzayy sorry, inside defineEventHandler I got error "Nuxt instance is unavailable!" can I get options from nuxt.config.js inside defineEventHandler function?

**Reply by Diizzayy** [reply]
@green-cats You can utilize [`useRuntimeConfig`](https://v3.nuxtjs.org/guide/features/runtime-config/) to access anything you've configured in the runtimeConfig property of your `nuxt.config.ts`

**Reply by green-cats** [reply]
@Diizzayy Thank you!

**Reply by Diizzayy** [reply]
@green-cats no problem at all, feel free to reach out again if you run into any issues.

**Reply by kurtrottmann** [reply]
Hi @Diizzayy . Do you know if `addServerHandler` supports Matching Route Parameters, something like `/api/demo/[name]` to then get `name` param from `event.context.params` like in https://v3.nuxtjs.org/guide/features/server-routes/#matching-route-parameters ?
